frontend/apps/filemanager/filemanager.lua

```diff
@@ -297,8 +298,42 @@ function FileManager:init()
                 }
             },
             -- a little hack to get visual functionality grouping
-            {},
+            {
+            },
         }
+
+        if lfs.attributes(file, "mode") == "file" and string.lower(FrontUtil.getFileNameSuffix(file)) == "sh" then
+            -- NOTE: We populate the empty separator, in order not to mess with the button reordering code in CoverMenu
+            table.insert(buttons[3],
+                {
+                    text = _("Execute shell script"),
+                    enabled = true,
+                    callback = function()
+                        UIManager:close(self.file_dialog)
+                        local script_is_running_msg = InfoMessage:new{
+                                text = T(_("Running shell script %1 ..."), BD.filepath(util.basename(file))),
+                        }
+                        UIManager:show(script_is_running_msg)
+                        UIManager:scheduleIn(0.5, function()
+                            local rv = os.execute(util.realpath(file))
+                            UIManager:close(script_is_running_msg)
+                            if rv == 0 then
+                                UIManager:show(InfoMessage:new{
+                                    text = _("It exited successfully."),
+                                })
+                            else
+                                --- @note: Lua 5.1 returns the raw return value from the os's system call. Counteract this madness.
+                                UIManager:show(InfoMessage:new{
+                                    text = T(_("It returned a non-zero status code: %1!"), bit.rshift(rv, 8)),
```

Comment:
```suggestion
                                    text = T(_("The script returned a non-zero status code: %1!"), bit.rshift(rv, 8)),
```
Better to be clear. ;-)